tests/phpunit/tests/fonts/font-library/wpFontsDir.php

```diff
@@ -69,4 +69,39 @@ function set_new_values( $defaults ) {
 
 		$this->assertSame( static::$dir_defaults, $font_dir, 'The wp_get_font_dir() method should return the default values.' );
 	}
+
+	public function test_fonts_dir_filters_do_not_trigger_infinite_loop() {
+		/*
+		 * Naive filtering of uploads directory to return font directory.
+		 *
+		 * This emulates the approach a plugin developer may take to
+		 * add the filter when extending the font library functionality.
+		 */
+		add_filter( 'upload_dir', 'wp_apply_font_dir_filter' );
+
+		add_filter(
+			'upload_dir',
+			function ( $upload_dir ) {
+				static $count = 0;
+				++$count;
+				// It may be hit a couple of times, at five iterations assume an infinite loop.
+				$this->assertLessThan( 5, $count, 'Filtering uploads directory should not trigger infinite loop.' );
```

Comment:
Rather than perform an assertion each time, what about this?
```php
if ( $count >= 5 ) {
	$this->fail( 'Filtering the uploads directory triggered an infinite loop.' );
}
```

Reply:
Done in 5117eef291bd58a9bd1eb1c104aaf0c77b0e91c3